### ISSUE TITLE
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte

### ISSUE THREAD

**Issue by mark-mcm** [google-research/language#51]
I was just trying to get the baseline model working with the preprocessed training set but I run into an issue upon trying to evaluate the tiny-dev:

```
Traceback (most recent call last):
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/site-packages/natural_questions/eval_utils.py", line 264, in read_annotation_from_one_split
    for line in input_file:
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/gzip.py", line 374, in readline
    return self._buffer.readline(size)
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/_compression.py", line 68, in readinto
    data = self.read(len(byte_view))
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/gzip.py", line 463, in read
    if not self._read_gzip_header():
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/gzip.py", line 406, in _read_gzip_header
    magic = self._fp.read(2)
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/gzip.py", line 91, in read
    self.file.read(size-self._length+read)
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/site-packages/natural_questions/nq_eval.py", line 462, in <module>
    app.run(main)
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/site-packages/absl/app.py", line 299, in run
    _run_main(main, args)
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/site-packages/absl/app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/site-packages/natural_questions/nq_eval.py", line 425, in main
    FLAGS.gold_path, n_threads=FLAGS.num_threads)
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/site-packages/natural_questions/eval_utils.py", line 303, in read_annotation
    dict_list = pool.map(read_annotation_from_one_split, input_paths)
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/multiprocessing/pool.py", line 268, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/home/mmcmill/anaconda3/envs/tf_gpu/lib/python3.7/multiprocessing/pool.py", line 657, in get
    raise self._value
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
```

Is the tiny-dev sample file incorrect? I don't understand why this would happen otherwise.

**Comment by mark-mcm** [comment]
Changed to a python 2 environment and no longer had the same issue.

**Comment by mark-mcm** [comment]
Also found a fix for running it on python 3.7.

-Change line 260 in eval_util.py within the natural_questions package from:
`gzipped_input_file = open(gzipped_input_file)`
to
`gzipped_input_file = open(gzipped_input_file, 'rb')`

-You also have to change line 338 in nq_eval.py within the same natural_questions package from:
`for threshold, (precision, recall) in scores_to_stats.iteritems():`
to
`for threshold, (precision, recall) in scores_to_stats.items():`

-Repeat the same change for line 411 in the same file from:
`return dict([(prefix + k, v) for k, v in metrics.iteritems()])`
to
`return dict([(prefix + k, v) for k, v in metrics.items()])`
